domain/modelupgrade/params.go

```diff
@@ -0,0 +1,17 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package modelupgrade
+
+import (
+	"github.com/juju/juju/core/semversion"
+)
+
+// UpgradeModelParams holds the parameters used to upgrade a model.
+type UpgradeModelParams struct {
+	ControllerModelVersion semversion.Number
+	TargetVersion          semversion.Number
+	AgentStream            string
+	IgnoreAgentVersions    bool
+	DryRun                 bool
```

Comment:
This is a parameter to the api and can see how we use it. But looking at the domain code it actually makes no sense in my head. By using DryRun we are asking the system to essentially report any errors with the process before making things final.

That is what happens regardless. I am not sure this should bleed into the domain now without a good reason.

Reply:
Dry run is used to check that the upgrade could go ahead. ie the new binaries can be found etc etc. If all that succeeds, we need the option of dry run to not actually record the upgrade intent.